### PR TITLE
Use a snapped location instead of a raw location during off-route detection process.

### PR DESCRIPTION
**Pull request by MaximAlien**
Change off-route detection logic to use snapped location instead of raw location.

/cc @azarovalex

### CODE REVIEW

File: Sources/MapboxCoreNavigation/RouteController.swift [hunk 1]
```diff
@@ -311,27 +311,34 @@ open class RouteController: NSObject {
...
+        ?? DefaultBehavior.shouldRerouteFromLocation
         
         updateSpokenInstructionProgress(status: status, willReRoute: willReroute)
         updateVisualInstructionProgress(status: status)
         updateRoadName(status: status)
-        updateDistanceToIntersection(from: CLLocation(status.location))
+        updateDistanceToIntersection(from: snappedLocation)
         
         if willReroute {
             reroute(from: CLLocation(status.location), along: routeProgress)
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/RouteController.swift:336]:
```suggestion
            reroute(from: snappedLocation, along: routeProgress)
```

File: Sources/MapboxCoreNavigation/RouteController.swift [hunk 2]
```diff
@@ -311,27 +311,34 @@ open class RouteController: NSObject {
...
+               rawLocation: rawLocation,
+               upcomingRouteAlerts: status.upcomingRouteAlerts)
         
         updateIndexes(status: status, progress: routeProgress)
         updateRouteLegProgress(status: status)
-        let willReroute = !userIsOnRoute(location, status: status) && delegate?.router(self, shouldRerouteFrom: location)
-            ?? DefaultBehavior.shouldRerouteFromLocation
+        let willReroute = !userIsOnRoute(snappedLocation, status: status)
+        && delegate?.router(self, shouldRerouteFrom: snappedLocation)
+        ?? DefaultBehavior.shouldRerouteFromLocation
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/RouteController.swift:328]: Could you place brackets here as it's difficult to visually analyze this part? I.e.
```suggestion
        && (delegate?.router(self, shouldRerouteFrom: snappedLocation)
            ?? DefaultBehavior.shouldRerouteFromLocation)
```

File: Sources/MapboxCoreNavigation/RouteController.swift (hunk 1 above)
Comment (MaximAlien) [reply Sources/MapboxCoreNavigation/RouteController.swift:336]: Yep, missed that one, thanks.

**Review by ShanMa1991 — APPROVED** [review]

**Comment by jill-cardamon** [comment]
Is `BillingHandlerUnitTests.testTripPerWaypoint()` a known flaky test?

**Comment by MaximAlien** [comment]
> Is `BillingHandlerUnitTests.testTripPerWaypoint()` a known flaky test?

Yep, we keep getting occasional errors in Core. Hopefully we'll resolve them in other PRs.

**Pull request merged**